core/src/main/java/org/ironjacamar/core/CoreLogger.java

```diff
@@ -57,7 +92,82 @@
    @Message(id = 202, value = "SecurityContext setup failed since CallbackSecurity was null")
    public void securityContextSetupFailedCallbackSecurityNull();
 
-   // CORE (300)
+   // CONNECTION MANAGER LISTENER (300)
+
+   /**
+    * Registered a null handle for managedConnection
+    * @param managedConnection The managedConnection instance
+    */
+   @LogMessage(level = INFO)
+   @Message(id = 301, value = "Registered a null handle for managed connection: %s")
+   public void registeredNullHandleManagedConnection(Object managedConnection);
+
+   /**
+    * Unregistered handle that was not registered
+    * @param handle Unregistered handle
+    * @param managedConnection The managedConnection instance
+    */
+   @LogMessage(level = INFO)
+   @Message(id = 302, value = "Unregistered handle that was not registered: %s" +
+         " for managed connection: %s")
+   public void unregisteredHandleNotRegistered(Object handle, Object managedConnection);
+
+   /**
+    * Unregistered a null handle for managedConnection
+    * @param managedConnection The managedConnection instance
+    */
+   @LogMessage(level = INFO)
+   @Message(id = 303, value = "Unregistered a null handle for managed connection: %s")
+   public void unregisteredNullHandleManagedConnection(Object managedConnection);
+
+   /**
+    * Connection error occured
+    * @param cl AbstractConnectionListener instance
+    * @param t The exception
+    */
+   @LogMessage(level = WARN)
+   @Message(id = 305, value = "Connection error occured: %s")
+   public void connectionErrorOccured(Object cl, @Cause Throwable t);
+
+   /**
+    * Unknown Connection error occured
+    * @param cl AbstractConnectionListener instance
+    * @param t The exception
+    */
+   @LogMessage(level = WARN)
+   @Message(id = 306, value = "Unknown connection error occured: %s")
+   public void unknownConnectionErrorOccured(Object cl, @Cause Throwable t);
+
+   /**
+    * Notified of error on a different managed connection
+    */
+   @LogMessage(level = WARN)
+   @Message(id = 307, value = "Notified of error on a different managed connection")
+   public void notifiedErrorDifferentManagedConnection();
+
+   /**
+    * throwable from unregister connection
+    * @param t The exception
+    */
+   @LogMessage(level = INFO)
+   @Message(id = 311, value = "Throwable from unregister connection")
+   public void throwableFromUnregisterConnection(@Cause Throwable t);
+
+   /**
+    * Error while closing connection handle
+    * @param t The exception
+    */
+   @LogMessage(level = ERROR)
+   @Message(id = 312, value = "Error while closing connection handle")
+   public void errorWhileClosingConnectionHandle(@Cause Throwable t);
+
+   /**
+    * There is something wrong with the pooling
+    * @param t The exception
+    */
+   @LogMessage(level = ERROR)
+   @Message(id = 313, value = "There is something wrong with the pooling")
+   public void somethingWrongWithPooling(@Cause Throwable t);
```

Comment:
I'm not sure of this, just a hunch: when this message is logged, is there another place in the log that will show more details of what went wrong?
Something sounds a little too vague to me... so I would change a little bit the message:
"Unexpected failure when pooling" <- notice that I'm not too familiar with the code and maybe my suggestions should be ignored.